Sources/Plasma/PubUtilLib/plResMgr/plLocalization.cpp

```diff
@@ -112,70 +122,69 @@ plFileName plLocalization::GetLocalized(plFileName const &name)
     return IGetLocalized(name, fLanguage);
 }
 
-plFileName plLocalization::ExportGetLocalized(const plFileName& name, int lang)
+plFileName plLocalization::ExportGetLocalized(const plFileName& name, Language lang)
 {
-    plFileName localizedName = IGetLocalized(name, Language(lang+1));
+    if (lang == kEnglish) {
+        return {};
+    }
+    plFileName localizedName = IGetLocalized(name, lang);
     if (plFileInfo(localizedName).Exists())
         return localizedName;
 
-    return "";
+    return {};
 }
 
 ST::string plLocalization::LocalToString(const std::vector<ST::string>& localizedText)
 {
     ST::string_stream ss;
-    for (size_t i = 0; i < localizedText.size(); i++)
-    {
-        if (i > kNumLanguages - 1)
+    for (auto lang : plLocalization::GetAllLanguages()) {
+        if (lang >= localizedText.size()) {
             break;
-        ST::string langName = GetLanguageName((Language)i);
-        ss << '$' << langName.substr(0, 2) << '$' << localizedText[i];
+        } else if (localizedText[lang].empty()) {
+            continue;
+        }
+        ST::string langName = GetLanguageName(lang);
+        ss << '$' << langName.substr(0, 2) << '$' << localizedText[lang];
     }
     return ss.to_string();
 }
 
 std::vector<ST::string> plLocalization::StringToLocal(const ST::string& localizedText)
 {
     std::vector<ST::string> tags;
-    std::vector<int> tagLocs;
-    std::vector<int> sortedTagLocs;
+    std::vector<hsSsize_t> tagLocs;
+    std::vector<Language> sortedLangs;
     std::vector<ST::string> retVal;
-    int i;
-    for (i=0; i<kNumLanguages; i++)
-    {
-        ST::string langName = GetLanguageName((Language)i);
+    for (auto lang : plLocalization::GetAllLanguages()) {
+        ST::string langName = GetLanguageName(lang);
         ST::string tag = "$" + langName.substr(0, 2) + "$";
         tags.push_back(tag);
         tagLocs.push_back(localizedText.find(tag));
-        sortedTagLocs.push_back(i);
+        sortedLangs.push_back(lang);
         retVal.emplace_back();
     }
-    for (i=0; i<kNumLanguages-1; i++)
-    {
-        for (int j=i; j<kNumLanguages; j++)
-        {
-            if (tagLocs[sortedTagLocs[i]] > tagLocs[sortedTagLocs[j]])
-                sortedTagLocs[i]^=sortedTagLocs[j]^=sortedTagLocs[i]^=sortedTagLocs[j]; // swap the contents (yes, it works)
-        }
-    }
-    // now sortedTagLocs has the indexes of tagLocs sorted from smallest loc to highest loc
+
+    std::sort(sortedLangs.begin(), sortedLangs.end(), [&](auto a, auto b) {
```

Comment:
```suggestion
    std::sort(sortedLangs.begin(), sortedLangs.end(), [&tagLocs](auto a, auto b) {
```

We prefer capture by name.